pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/HashJoinOperator.java

```diff
@@ -114,14 +125,98 @@ protected TransferableBlock getNextBlock() {
         return TransferableBlockUtils.getNoOpTransferableBlock();
       }
       // JOIN each left block with the right block.
-      return buildJoinedDataBlock(_leftTableOperator.nextBlock());
+      return buildJoinedDataBlock(getProbeOperator().nextBlock());
     } catch (Exception e) {
       return TransferableBlockUtils.getErrorTransferableBlock(e);
     }
   }
 
+  private Operator<TransferableBlock> getBroadcastOperator() {
```

Comment:
nit (suggestion): might be nicer to wrap all of these in a single Tuple class so that we can have only once switch statement:
```
static class JoinResolver {
  Operator<TransferableBlock> broadcastOperator;
  Operator<TransferableBlock> probeOperator;
  KeySelector broadcastSelector;
  KeySelector probOperator;
  BiFunction<Object[], Object[]> getLeftRow;
  BiFunction<Object[], Object[]> getRightRow;
}
```

and then just:
```
switch(_joinType) {
  case Left:
    return new JoinResolver(...)
}
```

Reply:
Done